app/src/main/java/de/westnordost/streetcomplete/tangram/QuestsMapFragment.java

```diff
@@ -99,6 +101,15 @@ public QuestsMapFragment()
 		}
 	}
 
+	@Override public void onResume()
+	{
+		super.onResume();
+		if (sceneFile != null && !sceneFile.equals(getSceneFilePath()))
+		{
+			getActivity().recreate();
+		}
+	}
+
```

Comment:
Why is this necessary? The user can only change the setting in another activity.
(Also, this code should be in the activity)

Reply:
Yes, I know, but the other activity (`MainActivity`) is not destroyed by switching to the settings activity, so the view is also not re-rendered when switching back from the settings activity to the main acitivity. But as it is stated here: https://blog.iamsuleiman.com/daynight-theme-android-tutorial-example/

> Calling `AppCompatDelegate.setDefaultNightMode(mode)` must be done BEFORE calling `super()` in Activity `onCreate()`. Otherwise you must `recreate()` the Activity for the set theme to take effect

I added this small check in order to recreate the activity if necessary...